static/js/narrow.js

```diff
@@ -148,6 +148,11 @@ exports.activate = function (raw_operators, opts) {
     if (raw_operators.length === 0) {
         return exports.deactivate();
     }
+    var known_operators = ["stream", "topic", "is", "pm-with", "sender", "subject"];
```

Comment:
Awesome, thanks for working on this @mohsenSy!

Cool, I like this general approach since it should be precise.  However, the list of operators isn't complete; there's also `in:`, `has:`, `near:`, and negated versions.  I think `filter.js` is the right place to create the canonical list of operators that we can use for this purpose, since that file is what we'll be updating whenever we add new operators (it has all the other code for handling the range of different narrow operators).

Having done a bit of research, I think that structurally, the best place to fix this would be to do this in the "FIXME" line in `Filter.parse`, since that's currently the single place where we parse search terms across the codebase, and we can then write a nice test for this behavior in `frontend_tests/node_tests/filter.js`.